src/System.Linq.Expressions/src/System/Linq/Expressions/IndexExpression.cs

```diff
@@ -383,52 +395,52 @@ private static void ValidateIndexedProperty(Expression instance, PropertyInfo in
                 ParameterInfo[] setParameters = setter.GetParametersCached();
                 if (setParameters.Length == 0)
                 {
-                    throw Error.SetterHasNoParams(nameof(indexer));
+                    throw Error.SetterHasNoParams(paramName);
                 }
 
                 // valueType is the type of the value passed to the setter (last parameter)
                 Type valueType = setParameters[setParameters.Length - 1].ParameterType;
                 if (valueType.IsByRef)
                 {
-                    throw Error.PropertyCannotHaveRefType(nameof(indexer));
+                    throw Error.PropertyCannotHaveRefType(paramName);
```

Comment:
Personally, I'd keep this - its more clear what exactly is wrong with the type, and also helps self-document this code